client/my-sites/backup/backup-contents-page/file-browser/file-info-card.tsx

```diff
@@ -59,7 +63,28 @@ const FileInfoCard: FunctionComponent< FileInfoCardProps > = ( { siteId, item }
 			} );
 	}, [ siteId, item, dispatch ] );
 
-	const showActions = item.type !== 'table' && item.type !== 'archive';
+	const prepareDownloadClick = useCallback( () => {
+		if ( ! item.period || ! fileInfo?.manifestFilter || ! fileInfo?.dataType ) {
+			// @TODO: We should dispatch an error notice
+			return;
+		}
+
+		prepareDownload( siteId, item.period, fileInfo.manifestFilter, fileInfo.dataType );
+	}, [ fileInfo, item.period, prepareDownload, siteId ] );
+
+	useEffect( () => {
+		if ( prepareDownloadStatus === PREPARE_DOWNLOAD_STATUS.PREPARING ) {
+			setIsDownloading( true );
```

Comment:
In the context of Preparing then downloading using the `IsDownloading` isn't super intuitive since it's preparing vs. downloading.  And here it looks like it would actually be set to false when the actual download starts.

It's reusing the same functionality for the spinner etc for the actual downloading, but I wonder if there's something more generic we could call it that fits both cases?

Reply:
> It's reusing the same functionality for the spinner etc for the actual downloading, but I wonder if there's something more generic we could call it that fits both cases?

I think it would be okay to call it `isPreparingDownload` or `isProcessingDownload` for both cases.
The difference between the first and the second is that we are waiting more time to get the URL to download the file.

What do you think?

Reply:
Agree. I will update it

Reply:
@jwebbdev Just updated it, could you take a look?